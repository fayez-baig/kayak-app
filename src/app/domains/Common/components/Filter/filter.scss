@import "../../../../../assets/css/index.scss";

.heading {
  margin: 67px 0 73px 0;
  font-weight: bold;
  font-size: 44px;
}

.subheading {
  color: $text-color-black;
  font-size: 16px;
  font-weight: bold;
}

.filter-wrapper {
  display: flex;
  margin: 23px 0 56px 0;
}
