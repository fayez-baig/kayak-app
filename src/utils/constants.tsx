export const ALLIANCES = [
  {
    code: "OW",
    name: "Oneworld",
  },
  {
    code: "ST",
    name: "Sky Team",
  },
  {
    code: "SA",
    name: "Star Alliance",
  },
];

export const ITEMS_PER_PAGE = 12;

export const INITIAL_PAGE_NUMBER = 1;

export const FILTER_TITLE = "Oneworld";

export const LOADER_TEST_ID = "suspense-loader";

export const ERROR_MESSAGE = "Something went wrong !!";

export const NEXT_BUTTON_TEST_ID = "next-btn";

export const PREV_BUTTON_TEST_ID = "prev-btn";

export const ERROR_BOUNDARY_MESSAGE = "Oops !";
