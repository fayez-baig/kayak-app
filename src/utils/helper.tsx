export const alliances = [
  {
    code: "OW",
    name: "Oneworld",
  },
  {
    code: "ST",
    name: "Sky Team",
  },
  {
    code: "SA",
    name: "Star Alliance",
  },
];

export const getFilteredData = (data, checkedValues) => {
  return !checkedValues.length
    ? data?.data
    : data?.data?.filter(({ alliance }) => checkedValues.includes(alliance));
};
