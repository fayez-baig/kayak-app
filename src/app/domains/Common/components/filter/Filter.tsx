import { FC, ReactElement } from "react";
import { ALLIANCES } from "utils/constants";
import { CheckBox } from "../ui";
import { FilterProps, HandleCheckedTypes } from "./types";
import { INITIAL_PAGE_NUMBER } from "utils/constants";

import "./filter.scss";

const Filter: FC<FilterProps> = ({
  checkedValues,
  setCheckedValues,
  setCurrentPage,
}): ReactElement => {
  const handleChecked = ({ target: { value } }: HandleCheckedTypes) => {
    setCurrentPage(INITIAL_PAGE_NUMBER);
    if (checkedValues.includes(value)) {
      const arr = checkedValues.filter((item) => item !== value);
      setCheckedValues(arr);
      return;
    }
    setCheckedValues([value, ...checkedValues]);
  };

  return (
    <>
      <h1 className="heading">Airlines</h1>
      <div className="subheading">Filter by Alliance</div>
      <div className="filter-wrapper">
        {ALLIANCES.map(({ code, name }) => (
          <CheckBox
            key={code}
            onChange={handleChecked}
            value={code}
            name={name}
            label={name}
          />
        ))}
      </div>
    </>
  );
};

export default Filter;
