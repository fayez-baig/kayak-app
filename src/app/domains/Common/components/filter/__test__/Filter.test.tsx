import { createRenderer } from "react-test-renderer/shallow";
import { render } from "@testing-library/react";
import fireEvent from "@testing-library/user-event";
import Filter from "../Filter";
import { FILTER_TITLE } from "utils/constants";

const renderer = createRenderer();

const props = {
  checkedValues: ["OW"],
  setCheckedValues: jest.fn(),
  setCurrentPage: jest.fn(),
};
describe("<Filter />", () => {
  it("should render and match the snapshot", () => {
    renderer.render(<Filter {...props} />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });

  it("checkbox should be unchecked", () => {
    const { getByTestId } = render(<Filter {...props} />);
    const input = getByTestId(FILTER_TITLE);
    fireEvent.click(input);
    expect(input).toBeTruthy();
  });

  it("checkbox should be checked", () => {
    props.checkedValues = ["SW"];
    const { getByTestId } = render(<Filter {...props} />);
    const input = getByTestId(FILTER_TITLE);
    fireEvent.click(input);
    expect(input).toBeTruthy();
  });
});
