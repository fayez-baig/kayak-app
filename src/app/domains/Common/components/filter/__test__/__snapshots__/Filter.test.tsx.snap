// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> should render and match the snapshot 1`] = `
<React.Fragment>
  <h1
    className="heading"
  >
    Airlines
  </h1>
  <div
    className="subheading"
  >
    Filter by Alliance
  </div>
  <div
    className="filter-wrapper"
  >
    <CheckBox
      label="Oneworld"
      name="Oneworld"
      onChange={[Function]}
      value="OW"
    />
    <CheckBox
      label="Sky Team"
      name="Sky Team"
      onChange={[Function]}
      value="ST"
    />
    <CheckBox
      label="Star Alliance"
      name="Star Alliance"
      onChange={[Function]}
      value="SA"
    />
  </div>
</React.Fragment>
`;
